fs/operations/operations.go

```diff
@@ -141,6 +141,40 @@ func Equal(ctx context.Context, src fs.ObjectInfo, dst fs.Object) bool {
 	return equal(ctx, src, dst, defaultEqualOpt(ctx))
 }
 
+// DirsEqual is like Equal but for dirs instead of objects.
+// It returns true if two dirs should be considered "equal" for the purposes of syncCopyMove
+// (in other words, true == "skip updating modtime/metadata for this dir".)
+// Unlike Equal, it does not consider size or checksum, as these do not apply to directories.
+func DirsEqual(ctx context.Context, src, dst fs.Directory, opt DirsEqualOpt) (equal bool) {
+	if dst == nil {
+		return false
+	}
+	ci := fs.GetConfig(ctx)
+	if ci.SizeOnly || ci.Immutable || ci.UseServerModTime || opt.ModifyWindow == fs.ModTimeNotSupported {
```

Comment:
I'm not sure `ci.UseServerModTime` is correct here? The destination could well do nothing with that flag (eg sync s3 -> local)?

Reply:
I think you're right, this one doesn't belong here. I think I was looking for something that does the oppose of `--ignore-times`, but that doesn't really exist.

I'm also thinking `ci.IgnoreExisting` probably ought to be added...right?

Reply:
Made this change.